graph_altimeter/scan/neptune_client.py

```diff
@@ -0,0 +1,151 @@
+# pylint: disable=arguments-differ
+"""This modules provides the class ``GraphAltimeterNeptuneClient`` that allows
+to write a graph generated by Altimeter as Security Graph universe snapshot."""
+
+import logging
+
+from gremlin_python.process.traversal import T
+from gremlin_python.process.graph_traversal import __
+from altimeter.core.neptune.client import (
+    AltimeterNeptuneClient,
+    NeptuneEndpoint,
+    NeptuneNoGraphsFoundException,
+    NeptuneLoadGraphException
+)
+
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+
+class GraphAltimeterNeptuneClient(AltimeterNeptuneClient):
+    """Client to run store the a graph generated by Altimeter into a Neptune
+    compatible database following the Security Graph Universe schema.
+
+    Args:
+        max_age_min: maximum acceptable age in minutes of graphs.  Only graphs
+        which are found
+                     that meet this critera will be queried.
+        neptune_endpoint: NeptuneEndpoint object for this client
+    """
+
+    def __init__(self, endpoint: NeptuneEndpoint):
+        super().__init__(0, endpoint)
+        self.logger = logger
+
+    def write_to_neptune_lpg(self, graph, scan_id, vertex_id):
+        """Write a graph generated by an Altimeter scan to a Neptune
+        database, ensuring that whenever a vertex is written, it is liked
+        to the given vertex id."""
+        if "vertices" in graph and "edges" in graph \
+                and len(graph["vertices"]) > 0:
+            g, conn = self.connect_to_gremlin()
+            self.__write_vertices(g, graph["vertices"], scan_id, vertex_id)
+            self.__write_edges(g, graph["edges"], scan_id)
+            conn.close()
```

Comment:
Should we use try-except-finally here to guarantee that the connection is properly closed?